src/sdk/object_io.js

```diff
@@ -467,7 +467,7 @@ class ObjectIO {
                     start: params.start,
                     end: params.start + chunk_info.size,
                     seq: params.seq,
-                    uncommitted: !params.complete_upload,
+                    uncommitted: !params.complete_upload && (chunks.length > 1 || typeof params.multipart_id !== 'undefined'),
```

Comment:
why do we care how many chunks? if this is not multipart then we can commit all parts.